src/pages/Order.tsx

```diff
@@ -0,0 +1,341 @@
+        수량
+      </ReceiverItemNumInputLabel>
+      <ReceiverItemNumInput
+        placeholder="수량"
+        value={itemCount}
+        onChange={(e) => setItemCount(parseInt(e.target.value))}
+      ></ReceiverItemNumInput>
+    </ReceiverInputWrapperStyle>
+  );
+}
+
+const ReceiverInputNameLabel = styled.label``;
+
+const ReceiverInputName = styled.input``;
+
+const ReceiverInputPhoneNumberLabel = styled.label``;
+
+const ReceiverInputPhoneNumber = styled.input``;
+
+const ReceiverItemNumInputLabel = styled.label``;
+
+const ReceiverItemNumInput = styled.input``;
+
+const ItemInfoTitle = styled.h2`
+  font-size: ${({ theme }) => theme.typography.title2Bold.fontSize};
+  font-weight: ${({ theme }) => theme.typography.title2Bold.fontWeight};
+  line-height: ${({ theme }) => theme.typography.title2Bold.lineHeight};
+  margin-bottom: 10px;
+`;
+
+const ItemInfoWrapperStyle = styled.div`
+  padding: 20px;
+`;
+
+function ItemInfoWrapper({ selectedItem }: { selectedItem: string | null }) {
+  const item = selectedItem ? JSON.parse(selectedItem) : null;
+
+  if (!item) return null;
+
+  return (
+    <ItemInfoWrapperStyle>
+      <ItemInfoTitle>상품 정보</ItemInfoTitle>
+      <img src={item.imageURL} alt="item.name" />
+      <h2>{item.name}</h2>
+      <p>{item.brandInfo.name}</p>
+      <p>{item.price.sellingPrice}원</p>
+    </ItemInfoWrapperStyle>
+  );
+}
+
+const OrderButtonStyle = styled.button`
+  background-color: ${({ theme }) => theme.colors.semantic.kakaoYellow};
+  width: 100%;
+  height: 50px;
+  border: none;
+`;
+
+interface ButtonProps {
+  receiverName: string;
+  // setReceiverName: React.Dispatch<React.SetStateAction<string>>;
+  receiverPhoneNum: string;
+  // setReceiverPhoneNum: React.Dispatch<React.SetStateAction<string>>;
+  itemCount: number;
+  // setItemCount: React.Dispatch<React.SetStateAction<number>>;
+}
+
+function OrderButton({
+  receiverName,
+  receiverPhoneNum,
+  itemCount,
+}: ButtonProps) {
+  const navigate = useNavigate();
+
+  function onClickHandler() {
+    let blocking = 0;
+
+    if (receiverName === '') {
+      blocking++;
+    }
+    if (!PHONE_NUM_REGEX.test(receiverPhoneNum)) {
+      blocking++;
+    }
+    if (itemCount < 1) {
+      blocking++;
+    }
+
+    if (blocking > 0) {
+      return 0;
+    }
+
+    alert('주문이 완료되었습니다.');
+    navigate('/');
+  }
+
+  return (
+    <OrderButtonStyle onClick={onClickHandler}>
+      {29000 * itemCount}원 주문하기
+    </OrderButtonStyle>
+  );
+}
+
+function Order() {
+  const [selectedId, setSelectedId] = useState<number | null>(null);
+  const [selectedCardImg, setSelectedCardImg] = useState(orderCard[0].imageUrl);
+  const [selectedCardTxt, setSelectedCardTxt] = useState(
+    orderCard[0].defaultTextMessage,
+  );
+
+  const [senderName, setSenderName] = useState('');
+
+  const [receiverName, setReceiverName] = useState('');
+  const [receiverPhoneNum, setReceiverPhoneNum] = useState('');
+  const [itemCount, setItemCount] = useState(1);
```

Comment:
상품 주문 폼에서의 입력값들과 검증 로직이 이리저리 흩어져 있는데 useOrderForm이라는 커스텀 훅으로 관리되면 좋을 것 같아요.

일단 아래처럼 간단하게 리팩터링부터 시작해볼까요?

```suggestion
  const {
    selectedId,
    senderName,
    receiverName,
    receiverPhoneNum,
    itemCount,
    handleChangeSelectedId,
    handleChangeSenderName,
    handleChangeReceiverName,
    handleChangeReceiverPhoneNum,
    handleChangeItemCount,
  } = useOrderForm();
```

Reply:
혹시 이 부분은 step3 진행하면서 리팩터링 해도될까요...?
제 능력부족, 시간관리를 잘 못해서...